### ISSUE TITLE
Blocked loading mixed active content in Issue view/edit

### ISSUE THREAD

**Issue by ashbike** [dhanasingh/redmine_wktime#6]
My Redmine is running over HTTPS and when I go to Issue details or edit page, It seems an Ajax call is failing. This apparently results in the Ajax 'Loading..." box continuously showing in Chrome. Firefox complains but the Loading box vanishes. They show the following error messages:

#### Firefox
Blocked loading mixed active content "http://redmine.bluecoppertech.com/wktime/getStatus?startDate=2014-11-21"[Learn More]
#### Google Chrome
Mixed Content: loaded over HTTPS, but requested an insecure XMLHttpRequest end... points to the URL http://redmine.bluecoppertech.com/wktime/getStatus?startDate=2014-11-21

A bit digging reveals that the URL being generated was using HTTP, not HTTPS. I just added ":only_path=>true" in the url_for call and it went away. However, I'm not a Ruby developer so I do not know the side-effects or if this is the way to fix it. I'm submitting the following for your review.

```
index a7968c7..51db147 100644
--- a/init.rb
+++ b/init.rb
@@ -123,7 +123,7 @@ class WktimeHook < Redmine::Hook::ViewListener
                wktime_helper = Object.new.extend(WktimeHelper)
                host_with_subdir = wktime_helper.getHostAndDir(req)
                "<div id='divError'><font color='red'>#{l(:label_warning_wktime_time_entry)}</font>
-                       <input type='hidden' id='getstatus_url' value='#{url_for(:controller => 'wktime', :action => 'getStatus',:host => host_with_subdir)}'>
+                       <input type='hidden' id='getstatus_url' value='#{url_for(:controller => 'wktime', :action => 'getStatus',:host => host_with_subdir, :only_path=>true)}'>
                </div>"
        end
```
### Redmine Information
Environment:
  Redmine version                2.5.3.stable.13623
  Ruby version                   1.9.3-p484 (2013-11-22) [x86_64-linux]
  Rails version                  3.2.19
  Environment                    production
  Database adapter               Mysql2
SCM:
  Subversion                     1.6.11
  Git                            1.7.1
  Filesystem                     
Redmine plugins:
  issuefy                        0.2.0
  progressive_projects_list      1.0.0
  redmine_banner                 0.1.0
  redmine_checklists             3.0.1
  redmine_ckeditor               1.0.16
  redmine_dashboard              2.2.4
  redmine_login_audit            0.1.5
  redmine_maintenance_mode       1.1.0
  redmine_spent_time_column      2.0.1
  redmine_tags                   2.0.1-dev
  redmine_wktime                 1.7
  sidebar_hide                   0.0.7

**Comment by dhanasingh** [comment]
fixed in 1.8

**Comment by kayrus** [comment]
Can you merge this fix for 1.7 (redmine 2.6)? Or probably create specific branch for redmine 2.6?

**Comment by suganya-thulasiraman** [comment]
We only support new features on the latest version, But this is a simple fix you can apply it to yourself.
Just add a parameter protocol in url_for in method "showWarningMsg" in file init.rb
Here is the fix,
#{url_for(:controller => 'wktime', :action => 'getStatus',:host => host_with_subdir,:protocol => req.protocol)}
